src/main/java/org/primeframework/mvc/security/DefaultEncryptor.java

```diff
@@ -16,53 +16,101 @@
 package org.primeframework.mvc.security;
 
 import javax.crypto.Cipher;
+import java.security.GeneralSecurityException;
 import java.security.SecureRandom;
 import java.util.Arrays;
 
 import com.google.inject.Inject;
+import com.google.inject.name.Named;
 
 /**
  * @author Daniel DeGroff
  */
 public class DefaultEncryptor implements Encryptor {
-  private final CipherProvider cipherProvider;
+  private final CipherProvider cbcCipherProvider;
+
+  private final CipherProvider gcmCipherProvider;
 
   @Inject
-  public DefaultEncryptor(CipherProvider cipherProvider) {
-    this.cipherProvider = cipherProvider;
+  public DefaultEncryptor(@Named("CBC") CipherProvider cbcCipherProvider, @Named("GCM") CipherProvider gcmCipherProvider) {
+    this.cbcCipherProvider = cbcCipherProvider;
+    this.gcmCipherProvider = gcmCipherProvider;
   }
 
   @Override
   public byte[] decrypt(byte[] bytes) throws Exception {
-    // The first 16 bytes are the IV
-    byte[] iv = new byte[16];
-    System.arraycopy(bytes, 0, iv, 0, 16);
-
-    Cipher cipher = cipherProvider.getDecryptor(iv);
+    // The first 16 bytes contain the initialization vector (IV)
+    byte[] iv = Arrays.copyOfRange(bytes, 0, 16);
+    // The remainder contains the encrypted bytes
+    byte[] encryptedBytes = Arrays.copyOfRange(bytes, 16, bytes.length);
 
-    byte[] json = Arrays.copyOfRange(bytes, 16, bytes.length);
-    byte[] result = new byte[cipher.getOutputSize(json.length)];
-    int resultLength = cipher.update(json, 0, result.length, result, 0);
-    resultLength += cipher.doFinal(result, resultLength);
-    return Arrays.copyOfRange(result, 0, resultLength);
+    try {
+      // Attempt to decrypt using AES/GCM
+      Cipher cipher = gcmCipherProvider.getDecryptor(iv);
+      return doDecrypt(encryptedBytes, cipher);
+    } catch (GeneralSecurityException gcmException) {
+      // If GCM failed, try decrypting in CBC mode
+      try {
+        Cipher cipher = cbcCipherProvider.getDecryptor(iv);
+        return doDecrypt(encryptedBytes, cipher);
+      } catch (GeneralSecurityException cbcException) {
+        // If CBC also failed, re-throw the original GCM exception
+        throw gcmException;
+      }
+    }
   }
 
   @Override
   public byte[] encrypt(byte[] bytes) throws Exception {
-    // The first 16 bytes are the IV
-    byte[] iv = new byte[16];
-    new SecureRandom().nextBytes(iv);
+    byte[] iv = generateIV();
```

Comment:
If this is only used in this one place, we should just inline it. It will be faster.

Reply:
1. If we're worrying about the overhead to call a function, we are using the wrong language.
2. It's probably optimized out by the compiler anyway.

Reply:
I agree, it may be optimized out. 

However, all changes we make to public libraries should be as pristine as possible. In general we don't want to make methods that are only called once. 

The code used to be two lines, and you have 5 line function to replace it. Please revert.